From 1abe96527ce323b5295499d45ca46335e358c867 Mon Sep 17 00:00:00 2001
From: Thomas Gleixner <tglx@linutronix.de>
Date: Thu, 3 Mar 2016 12:24:10 +0100
Subject: cpu/hotplug: Remove redundant state check
Reply-To: joseph.salisbury@microsoft.com

The check for the AP range in cpuhp_is_ap_state() is redundant after commit
8df3e07e7f21 "cpu/hotplug: Let upcoming cpu bring itself fully up" because all
states above CPUHP_BRINGUP_CPU are invoked on the hotplugged cpu. Remove it.

Reported-by: Richard Cochran <richardcochran@gmail.com>
Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
(cherry picked from commit 5d8eb84253333f8f63ec704276e3f0a8ec8f3189)
Signed-off-by: Joseph Salisbury <joseph.salisbury@microsoft.com>
---
 kernel/cpu.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/kernel/cpu.c b/kernel/cpu.c
index de4c3277f7d6..d0fb7e080a6f 100644
--- a/kernel/cpu.c
+++ b/kernel/cpu.c
@@ -1238,8 +1238,6 @@ static int cpuhp_cb_check(enum cpuhp_state state)
 
 static bool cpuhp_is_ap_state(enum cpuhp_state state)
 {
-	if (state >= CPUHP_AP_OFFLINE && state <= CPUHP_AP_ONLINE)
-		return true;
 	return state > CPUHP_BRINGUP_CPU;
 }
 
-- 
2.17.1

